packages/rum-core/src/domain/rumEventsCollection/action/trackClickActions.ts

```diff
@@ -252,7 +253,13 @@ function newClick(
   const id = generateUUID()
   const startClocks = clocksNow()
   const historyEntry = history.add(id, startClocks.relative)
-  const eventCountsSubscription = trackEventCounts(lifeCycle)
+  const eventCountsSubscription = trackEventCounts({
+    lifeCycle,
+    predicate: (event) =>
+      event.type !== 'view' &&
+      event.action !== undefined &&
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** FMU we are excluding `view` type because of `action.count`‏ attribute, should we directly have a check like:
```ts
event.action?.id !== undefined
```

Reply:
We are excluding `view` because on `RumViewEvent`, the `action` property has type `unknown` ([any undefined property has type `unknown`](https://github.com/DataDog/browser-sdk/blob/911084ad512096420f790f9e25ebc3ba255610e7/packages/rum-core/src/rumEvent.types.ts#L750)). So we can't do `event.action?.id` directly.

TS v4.9 improve ["unknown" type narrowing](https://devblogs.microsoft.com/typescript/announcing-typescript-4-9/#unlisted-property-narrowing-with-the-in-operator) and we'll be able to do something [like that](https://www.typescriptlang.org/play?ts=5.0.0-dev.20221125#code/JYOwLgpgTgZghgYwgAgEoFcC2A1YEDuAogG4TjIDeAUMrcgNoDWAXMgM5hSgDmAuq+hCMQAe3wgqAXyqhIsRCgyZCUKCKgkyYSjTqIwwESAD8rCsmAATVhy4huyadLABPAA6Ksm8gF40X1XVvbQAffxw8IlJwKioYQQQDI2Q3KAhLYAQ4SAAKCGiwViVggEodWjSwdCgQZFcPERhkfK0AOn1DWp9u5AAiEQAjACsIRN7kADIJ5oL2xM7kAEIekHQAGzXJ6d6rcdAZto7kqYPwOaSQVqtkbr9esBEH3qkgA), but for now the easiest thing is to exclude types that don't define an `action` property